### ISSUE TITLE
Accuracy incurs a GPU/CPU sync point

### ISSUE THREAD

**Issue by ndrwrbgs** [Lightning-AI/torchmetrics#2638]
## 🐛 Bug

If running Accuracy inside the training loop, training is slowed due to CPU/GPU sync point

### To Reproduce

Steps to reproduce the behavior...

Pull Request includes reproduction, a unit test which fails before the modification and passes after.
[#1 Update _safe_divide to allow Accuracy to run on the GPU](https://github.com/ndrwrbgs/torchmetrics/pull/1)

_P.S. I opened it in the fork due to the question # 1 under "Before submitting", but [Contributing.md](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md) does not mandate discussion prior "Submit a PR!", would it be possible to update one or the other with the latest guidance?_

### Expected behavior

Accuracy can be calculated on the GPU, without syncing

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): conda    torchmetrics              1.4.0.post0        pyhd8ed1ab_0    conda-forge
- Python & PyTorch Version (e.g., 1.0):  python                    3.11.9          hb806964_0_cpython    conda-forge
- Any other relevant information such as OS (e.g., Linux):

**Comment by SkafteNicki** [comment]
Hi @ndrwrbgs, thanks for reporting this issue and thanks for already having a fix and unittest for the bug!
Could you open a PR directly to the main torchmetrics repository so we can get the fix incorporated?

**Comment by ndrwrbgs** [comment]
Thank you, all done at #2640
